packages/integration/lib/src/views/poll/handler/poll_handler.dart

```diff
@@ -0,0 +1,366 @@
+        0;
+    for (int i = 0; i < options.length; i++) {
+      int index = attachmentMeta.options!
+          .indexWhere((element) => element.id == options[i]);
+      if (index != -1) {
+        attachmentMeta.options![index].isSelected = false;
+        attachmentMeta.options![index].voteCount--;
+        attachmentMeta.options![index].percentage =
+            (attachmentMeta.options![index].voteCount / totalVotes) * 100;
+      }
+    }
+
+    attachmentMeta = previousValue;
+    resetOptions(attachmentMeta, previousValue);
+    LMFeedCore.showSnackBar(
+      context,
+      e.toString(),
+      source,
+    );
+  }
+}
+
+void resetOptions(LMAttachmentMetaViewData attachmentMeta,
+    LMAttachmentMetaViewData previousValue) {
+  attachmentMeta = previousValue;
+}
+
+bool showTick(
+    LMAttachmentMetaViewData attachmentMeta,
+    LMPollOptionViewData option,
+    List<String> selectedOption,
+    bool isVoteEditing) {
+  // if (isPollSubmitted(attachmentMeta.options!)) {
+  //   return false;
+  // }
+  if (isVoteEditing) {
+    return selectedOption.contains(option.id);
+  }
+  if ((isMultiChoicePoll(attachmentMeta.multiSelectNo!,
+                  attachmentMeta.multiSelectState!) ==
+              true ||
+          isInstantPoll(attachmentMeta.pollType!) == false) &&
+      option.isSelected == true) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+bool showAddOptionButton(LMAttachmentMetaViewData attachmentMeta) {
+  bool isAddOptionAllowedForInstantPoll =
+      isInstantPoll(attachmentMeta.pollType) &&
+          !isPollSubmitted(attachmentMeta.options!);
+  bool isAddOptionAllowedForDeferredPoll =
+      !isInstantPoll(attachmentMeta.pollType);
+
+  if (attachmentMeta.allowAddOption != null &&
+      attachmentMeta.allowAddOption! &&
+      !hasPollEnded(attachmentMeta.expiryTime) &&
+      (isAddOptionAllowedForInstantPoll || isAddOptionAllowedForDeferredPoll)) {
+    if (attachmentMeta.options!.length < 10) return true;
+  }
+  return false;
+}
+
+bool showSubmitButton(LMAttachmentMetaViewData attachmentMeta) {
+  if (isPollSubmitted(attachmentMeta.options!)) {
+    return false;
+  }
+  if ((attachmentMeta.pollType != null &&
+          isInstantPoll(attachmentMeta.pollType!) &&
+          isPollSubmitted(attachmentMeta.options!)) ||
+      hasPollEnded(attachmentMeta.expiryTime)) {
+    return false;
+  } else if (!isMultiChoicePoll(
+      attachmentMeta.multiSelectNo, attachmentMeta.multiSelectState)) {
+    return false;
+  } else
+    return true;
+}
+
+Future<void> addOption(
+    BuildContext context,
+    LMFeedPoll pollWidget,
+    LMAttachmentMetaViewData attachmentMeta,
+    String option,
+    String postId,
+    LMUserViewData? currentUser,
+    ValueNotifier<bool> rebuildPostWidget,
+    LMFeedWidgetSource source) async {
+  if ((pollWidget.attachmentMeta.options?.length ?? 0) > 10) {
+    LMFeedCore.showSnackBar(
+      context,
+      "You can add maximum 10 options",
```

Comment:
Get this reviewed with @animeshdas-lm , the string should be "You can add a maximum of 10 options" acc to me

Reply:
accroding to him it should be "You can add upto 10 options." changed it to the same

Reply:
Sorted